plansys2_problem_expert/src/plansys2_problem_expert/ProblemExpert.cpp

```diff
@@ -304,6 +307,7 @@ ProblemExpert::existPredicate(const Predicate & predicate)
       predicates_[i].parameters == predicate.parameters)
     {
       found = true;
+      break; // returning here speeds up this method
```

Comment:
The same here.

Reply:
Same behavior should be added to `existAssignment `(for behavior homogeneity)